app/models/my_module_group.rb

```diff
@@ -10,6 +10,10 @@ class MyModuleGroup < ApplicationRecord
              optional: true
   has_many :my_modules, inverse_of: :my_module_group, dependent: :nullify
 
+  scope :active, (lambda do
+    joins(:my_modules).where('my_modules.archived = ?', false).distinct
```

Comment:
Ok cool. Maybe only naming is a bit confused here. This is why I have been confused before and didn't get what I was looking on... 
We are scoping `my_modules_groups` but on `my_module` condition. 
What do you think to rename scope to something like `without_archived_modules`... 

Than `my_moudle_groups.without_archived_modules` will return you exactly what you read. 

Am I right and are you agree? :)

Reply:
:+1: